doc/source/train/examples/intel_gaudi/llama_deepspeed.ipynb

```diff
@@ -0,0 +1,2436 @@
+    "- Require an HPU for each worker in ScalingConfig\n",
+    "- Set backend to `hccl` in TorchConfig"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "def train_llama(num_workers, execution_mode):\n",
+    "    import ray\n",
+    "    from ray.train import ScalingConfig\n",
+    "    from ray.train.torch import TorchTrainer, TorchConfig\n",
+    "\n",
+    "    # deepspeed config, can also place it to config file\n",
+    "    deepspeed_config = {\n",
+    "        \"steps_per_print\": 64,\n",
+    "        \"train_batch_size\": \"auto\",\n",
+    "        \"train_micro_batch_size_per_gpu\": \"auto\",\n",
+    "        \"gradient_accumulation_steps\": \"auto\",\n",
+    "        \"bf16\": {\n",
+    "            \"enabled\": True\n",
+    "        },\n",
+    "        \"gradient_clipping\": 1.0,\n",
+    "        \"zero_optimization\": {\n",
+    "            \"stage\": 3,\n",
+    "            \"overlap_comm\": False,\n",
+    "            \"contiguous_gradients\": False,\n",
+    "            \"stage3_gather_16bit_weights_on_model_save\": True\n",
+    "        }\n",
+    "    }\n",
+    "\n",
+    "    # Preparing train configurations\n",
+    "    train_config = {\n",
+    "        \"execution_mode\": execution_mode,\n",
+    "        \"model\": \"/root/models/models--meta-llama--Llama-2-70b-chat-hf/snapshots/e9149a12809580e8602995856f8098ce973d1080/\",\n",
+    "        \"model_config\": {\"torch_dtype\": torch.bfloat16, \"trust_remote_code\": None, \"use_auth_token\": None},\n",
+    "        \"lora_config\": {\"task_type\": \"CAUSAL_LM\", \"r\": 4, \"lora_dropout\": 0.1, \"target_modules\": [\"q_proj\", \"v_proj\", \"k_proj\", \"o_proj\"]},\n",
+    "        \"lr\": 0.0018,\n",
+    "        \"epochs\": 2,\n",
+    "        \"batch_size_per_worker\": 10,\n",
+    "        \"output\": \"/tmp/ray/\",\n",
+    "        \"deepspeed\": deepspeed_config,\n",
+    "    }\n",
+    "\n",
+    "    # Configure computation resources\n",
+    "    # In ScalingConfig, require an HPU for each worker\n",
+    "    scaling_config = ScalingConfig(num_workers=num_workers,\n",
+    "                                   use_gpu=False,\n",
+    "                                   resources_per_worker={\"CPU\": 1, \"HPU\": 1})\n",
+    "    # Set backend to hccl in TorchConfig\n",
+    "    torch_config = TorchConfig(backend=\"hccl\")\n",
+    "\n",
+    "    runtime_env = {\n",
+    "        \"env_vars\": {\n",
+    "            \"PT_HPU_MAX_COMPOUND_OP_SIZE\": \"10\",\n",
+    "            \"DEEPSPEED_HPU_ZERO3_SYNC_MARK_STEP_REQUIRED\": \"1\",\n",
+    "            \"WORLD_SIZE\": str(num_workers),\n",
+    "            \"HABANA_VISIBLE_MODULES\": \"0,1,2,3,4,5,6,7\",\n",
+    "            \"RAY_EXPERIMENTAL_NOSET_HABANA_VISIBLE_MODULES\": \"true\",\n",
```

Comment:
Any particular reason to disable `RAY_EXPERIMENTAL_NOSET_HABANA_VISIBLE_MODULES`, and setup `WORLD_SIZE`, `HABANA_VISIBLE_MODULES` by yourself here? They should already be setup by Ray?

Reply:
Can you check if these 3 env var settings can be removed?

Reply:
yes, removed these 3 env var.